application/src/main/java/org/togetherjava/tjbot/commands/bytecode/BytecodeCommand.java

```diff
@@ -0,0 +1,269 @@
+import org.togetherjava.tjbot.imc.CompileInfo;
+import org.togetherjava.tjbot.imc.InMemoryCompiler;
+import org.togetherjava.tjbot.imc.JavacOption;
+import org.togetherjava.tjbot.javap.Javap;
+import org.togetherjava.tjbot.javap.JavapOption;
+
+import javax.tools.Diagnostic;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Bytecode command that uses the message context commands Example usage:
+ * <p>
+ *
+ * <pre>
+ * {@code
+ *     ```java
+ *     class Test {
+ *         public void foo() {
+ *             System.out.println("Hello World!");
+ *         }
+ *     }
+ *     ```
+ * }
+ * </pre>
+ *
+ * then apply the application
+ */
+public final class BytecodeCommand extends BotCommandAdapter
+        implements MessageContextCommand, MessageReceiver {
+    private static final String CODE_BLOCK_OPENING = "```\n";
+    private static final String CODE_BLOCK_CLOSING = "\n```";
+
+    private final Pattern codeBlockExtractorPattern =
+            Pattern.compile("```(?:java)?\\s*([\\w\\W]+)```|``?([\\w\\W]+)``?");
+    private final Map<Long, List<Long>> userToBotMessages = new HashMap<>();
+
+    public BytecodeCommand() {
+        super(Commands.message("View bytecode"), CommandVisibility.GUILD);
+    }
+
+    // Fresh compile when a user uses the message context command on one of their messages
+    @Override
+    public void onMessageContext(@NotNull MessageContextInteractionEvent event) {
+        Message message = event.getTarget();
+        String content = message.getContentRaw();
+
+        event.reply("Compiling...")
+            .mentionRepliedUser(false)
+            .flatMap(InteractionHook::retrieveOriginal)
+            .queue(compReply -> compile(message, compReply, parseCommandFromMessage(content)));
+    }
+
+    // Delete our messages if the user deletes their request message
+    @Override
+    public void onMessageDeleted(@NotNull MessageDeleteEvent event) {
+        long mesageIdLong = event.getMessageIdLong();
+
+        if (userToBotMessages.containsKey(mesageIdLong)) {
+            deleteMyMessages(mesageIdLong, event.getTextChannel());
+        }
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        /* empty */
+    }
+
+    // Recompile when the user sends edits their request message
+    @Override
+    public void onMessageUpdated(@NotNull MessageUpdateEvent event) {
+        Message message = event.getMessage();
+        long messageIdLong = event.getMessageIdLong();
+
+        if (!userToBotMessages.containsKey(messageIdLong)) {
+            return;
+        }
+
+        TextChannel textChannel = message.getTextChannel();
+        List<Long> myMessages = userToBotMessages.get(messageIdLong);
+
+        if (myMessages.isEmpty()) {
+            message.reply(
+                    "An unknown error occurred (`userMessagesToMyMessages.get(messageIdLong).isEmpty() == true`)")
```

Comment:
```suggestion
                    "An unknown error occurred (`userMessagesToBotMessages.get(messageIdLong).isEmpty() == true`)")
```
Forgot after renaming or intentional?

Reply:
forgor after renaming